frame/message-queue/src/lib.rs

```diff
@@ -1085,6 +1090,111 @@ impl<T: Config> Pallet<T> {
 		ItemExecutionStatus::Executed(is_processed)
 	}
 
+	/// Ensure the correctness of state of this pallet.
+	///
+	/// # Assumptions-
+	///
+	/// If `serviceHead` points to a ready Queue, then BookState of that Queue has:
+	///
+	/// * `message_count` > 0
+	/// * `size` > 0
+	/// * `end` > `begin`
+	/// * Some(ready_neighbours)
+	/// * If `ready_neighbours.next` == self.origin, then `ready_neighbours.prev` == self.origin
+	///   (only queue in ring)
+	///
+	/// For Pages(begin to end-1) in BookState:
+	///
+	/// * `remaining` > 0
+	/// * `remaining_size` > 0
+	/// * `first` <= `last`
+	/// * Every page can be decoded into peek_* functions
+
+	pub fn do_try_state() -> Result<(), &'static str> {
+		// Checking memory corruption for BookStateFor
+		assert_eq!(
+			BookStateFor::<T>::iter_keys().count(),
+			BookStateFor::<T>::iter_values().count(),
+			"Memory Corruption in BookStateFor"
+		);
+
+		// Checking memory corruption for Pages
+		assert_eq!(
+			Pages::<T>::iter_keys().count(),
+			Pages::<T>::iter_values().count(),
+			"Memory Corruption in Pages"
+		);
+
+		// No state to check
+		if ServiceHead::<T>::get().is_none() {
+			return Ok(())
+		}
+
+		//loop around this origin
+		let starting_origin = ServiceHead::<T>::get().unwrap();
+
+		while let Some(head) = Self::bump_service_head(&mut WeightMeter::max_limit()) {
+			assert!(
```

Comment:
This should actually be `ensure!` here and down below - since the function returns an error.  
Sorry for not noticing earlier :see_no_evil:   

Please put your address in the MR description for a tip.